carbon-design-system/carbon
# revert(button): revert changes to button for v11

Closes #10505 
Closes #10294 

This PR reverts the changes we made to button for icon support. This should address most of the issues with components using `Button` as an icon button.

#### Changelog

**New**

**Changed**

- Revert changes to button to mirror v10 behavior

**Removed**

#### Testing / Reviewing

- Verify issues identified in referenced issues have been resolved
- Verify `IconButton` still works as intended
